In [1]:
import pandas as pd

In [2]:
df = pd.read_csv("/Users/patrickburke/Library/CloudStorage/OneDrive-EmoryUniversity/ECON496RW/cleaned_small/factpcradditionalresponsemode.csv")

In [3]:
df.head(20)

PcrKey  eResponse_24
0     1472192       7701003
1    13273412       2224019
2    14548133       2224015
3    17184298       2224015
4    21699657       2224019
5    21719864       7701003
6    72057010       2224019
7    73335292       2224015
8    79846935       2224019
9    81481800       7701003
10   81954292       7701003
11   83733589       2224015
12   88178354       2224019
13   97455081       2224019
14  102488422       2224015
15  107295362       2224015
16  110454483       7701003
17  113861204       2224015
18  116400273       2224015
19  118123742       2224015

In [6]:
len(df.index) #bruhhh 60,716,873 what the hell

60716873

In [7]:
# check for duplicates
dup_check = df.duplicated(subset="PcrKey",keep='first')
dup_check

0           False
1           False
2           False
3           False
4           False
            ...  
60716868    False
60716869    False
60716870    False
60716871    False
60716872    False
Length: 60716873, dtype: bool

In [8]:
print(dup_check.value_counts()[True])

11733883


In [9]:
dup_check = df.duplicated(subset="PcrKey",keep=False)
dup_df = df[dup_check]

In [12]:
dup_df.head()

PcrKey  eResponse_24
20  118150307       2224003
21  118150307       2224009
22  118150307       2224013
23  118150307       2224005
24  118150307       2224015

In [13]:
dup_df.loc[dup_df['PcrKey'] == 118150307]

PcrKey  eResponse_24
20  118150307       2224003
21  118150307       2224009
22  118150307       2224013
23  118150307       2224005
24  118150307       2224015

- 2224001 - Intersection Navigation-Against Normal Light Patterns
- 2224003 - Intersection Navigation-With Automated Light Changing Technology 
- 2224005 - Intersection Navigation-With Normal Light Patterns
- 2224007 - Scheduled
- 2224009 - Speed-Enhanced per Local Policy
- 2224011 - Speed-Normal Traffic
- 2224013 - Unscheduled
- 2224015 - Lights and Sirens
- 2224017 - Lights and No Sirens
- 2224019 - No Lights or Sirens
- 2224021 - Initial No Lights or Sirens, Upgraded to Lights and Sirens
- 2224023 - Initial Lights and Sirens, Downgraded to No Lights or Sirens

- Intersection Navigation-With Automated Light Changing Technology, Speed-Enhanced per Local Policy, Unscheduled, Intersection Navigation-With Normal Light Patterns, Lights and Sirens
- Only conflict here is multiple types of intersection navigation but thats actually very possible, duplicates OK